contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -1836,6 +1836,82 @@ check_fulltext_exist(const char *schema_name, const char *table_name)
 	return ft_index_name != NULL;
 }
 
+/*
+ * replace_special_chars_fts_impl
+ * Replace special characters in the input string with the unique hash
+ */
+char
+*replace_special_chars_fts_impl(char *input_str) {
+	size_t		input_len = strlen(input_str);
+	char		*replacement = NULL;
+	char		*result;
+	char		*unique_hashes[4];
+	const char	*special_chars[4] = {"~!&|@#$%^*+=\\;:<>?./", "`", "'", "_"};
+	StringInfoData	output_str;
+	
+	for (int i = 0; i < 4; i++) {
+		unique_hashes[i] = construct_unique_index_name("specialChars", psprintf("cat%d", i + 1));
+	}
+	
+	initStringInfo(&output_str);
+
+	for (size_t i = 0; i < input_len; i++) {
+		replacement = NULL;
+
+		for (int j = 0; j < 4; j++) {
+			if (strchr(special_chars[j], input_str[i]) != NULL) {
+				replacement = unique_hashes[j];
+				break;
+			}
+		}
+
+		if (replacement != NULL) {
+			/* Remove leading and trailing spaces around special characters */
+			size_t start = i;
+			size_t end = i;
+
+			/* Look for leading spaces */
+			while (start > 0 && isspace((unsigned char)input_str[start - 1])) {
```

Comment:
This clause basically removes leading and trailing spaces around special characters. Essentially, strings like `"one@two"` and `"one @ two"` these should be treated as same, so that's why adding this clause to trim spaces.